test/parallel/test-worker-dns-terminate.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+const common = require('../common');
+const { Worker } = require('worker');
+
+const w = new Worker(`
+const dns = require('dns');
+dns.lookup('nonexistent.org', () => {});
+require('worker').postMessage('0');
+`, { eval: true });
+
+w.on('message', common.mustCall(() => {
+  w.terminate(common.mustCall());
+}));
```

Comment:
What exactly is this file testing? Maybe a comment explaining why this is needed?

Reply:
I guess it’s trying to tear down a worker that’s currently performing a DNS request? Just making sure nothing crashes? It’s been a few weeks since I wrote this myself, you can safely assume I know as much about the code as you do :smile:

Reply:
Anyway, I’ve added a comment here :)